mainwindow.cpp

```diff
@@ -1,14 +1,69 @@
 #include "mainwindow.h"
-#include "./ui_mainwindow.h"
+#include "ui_mainwindow.h"
+#include <QDebug>
 
+// Toevoegen voor toegang tot de structs
+extern "C" {
+#include "packets.h"
+}
+
+// Constructor van MainWindow
 MainWindow::MainWindow(QWidget *parent)
-    : QMainWindow(parent)
-    , ui(new Ui::MainWindow)
-{
+    : QMainWindow(parent), ui(new Ui::MainWindow) {
+
+    // Koppelt de UI-bestand aan deze class
     ui->setupUi(this);
+
+    // RGB-sliders instellen op bereik 0–255
+    ui->sliderR->setRange(0, 255);
+    ui->sliderG->setRange(0, 255);
+    ui->sliderB->setRange(0, 255);
 }
 
-MainWindow::~MainWindow()
-{
+// Destructor: zorgt ervoor dat geheugen van de UI wordt opgeruimd
+MainWindow::~MainWindow() {
     delete ui;
 }
+
+// Wordt uitgevoerd als op 'Stel kleur in' wordt geklikt
+void MainWindow::on_btnSetColor_clicked() {
+    int r = 0, g = 0, b = 0;
+
+    if (ui->checkBoxLightOn->isChecked()) {
+        r = ui->sliderR->value();
+        g = ui->sliderG->value();
+        b = ui->sliderB->value();
+    }
+
+    // Pas de kleur toe in de GUI
+    applyLightColor(r, g, b);
+
+    // Bouw het sensor_packet voor RGB
+    sensor_packet pakket;
+    pakket.header.ptype = PacketType::DASHBOARD_POST;
+    pakket.header.length = sizeof(sensor_header) + sizeof(sensor_packet_rgb_light);
+
+    pakket.data.rgb_light.metadata.sensor_type = SensorType::RGB_LIGHT;
+    pakket.data.rgb_light.metadata.sensor_id = 1;
```

Comment:
Kunnen we hiervoor een settings ding maken zodat de sensor niet permanent gehardcode is